packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -2437,3 +2437,26 @@ abstract class PopEntry<T> {
     return 'PopEntry canPop: ${canPopNotifier.value}, onPopInvoked: $onPopInvokedWithResult';
   }
 }
+
+/// Mixin for a route that can provide a delegated secondary transition to the
+/// outgoing route.
+mixin FlexibleTransitionRouteMixin<T> on TransitionRoute<T> {
```

Comment:
I like this name.

So I guess existing routes will not mix this in and so will not be flexible by default. That's probably a good thing. Also, I'm thinking it's probably rare for people to create their own route outside of the framework.